### PR TITLE
[PM-13073] Handle Fido2 credential errors on vault unlock screen

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13073

## 📔 Objective

This commit adds logic to handle Fido2 credential errors on the vault
unlock screen.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt [hunk 1]
```diff
@@ -133,6 +139,17 @@ class VaultUnlockViewModel @Inject constructor(
 
     private fun handleDismissDialog() {
         mutableStateFlow.update { it.copy(dialog = null) }
+        when {
+            state.fido2GetCredentialsRequest != null -> {
+                sendEvent(VaultUnlockEvent.Fido2GetCredentialsError)
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt:144]: Hmmm, is this supposed to be here?

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt [hunk 2]
```diff
@@ -389,6 +411,7 @@ data class VaultUnlockState(
          */
         @Parcelize
         data class Error(
+            val title: Text,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt:414]: 👍

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt (hunk 1 above)
Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt:144]: Yes. When the dialog is dismissed and we have a FIDO2 request in state we have to notify Android an error occurred. This event is how we're triggering the appropriate error response.

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt [hunk 3]
```diff
@@ -108,6 +113,17 @@ fun VaultUnlockScreen(
...
+            VaultUnlockEvent.Fido2CredentialAssertionError -> {
+                fido2CompletionManager.completeFido2Assertion(
+                    result = Fido2CredentialAssertionResult.Error,
+                )
+            }
+            VaultUnlockEvent.Fido2GetCredentialsError -> {
+                fido2CompletionManager.completeFido2GetCredentialRequest(
+                    result = Fido2GetCredentialsResult.Error,
+                )
+            }
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:126]: Do we need to set the `title = dialog.title` below?

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:126]: Yes! Thank you.

**Review by dseverns-livefront — APPROVED** [review]

**Pull request merged**